test/functional/core/main_spec.lua

```diff
@@ -0,0 +1,130 @@
+local lfs = require('lfs')
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+
+local eq = helpers.eq
+local feed = helpers.feed
+local eval = helpers.eval
+local clear = helpers.clear
+local funcs = helpers.funcs
+local nvim_prog = helpers.nvim_prog
+local write_file = helpers.write_file
+
+local function nvim_prog_abs()
+  -- system(['build/bin/nvim']) does not work for whatever reason. It needs to
+  -- either be executable searched in $PATH or something starting with / or ./.
+  if nvim_prog:match('[/\\]') then
+    return funcs.fnamemodify(nvim_prog, ':p')
+  else
+    return nvim_prog
+  end
+end
+
+describe('Command-line option', function()
+  describe('-s', function()
+    local fname = 'Xtest-functional-core-main-s'
+    local fname_2 = fname .. '.2'
+    local nonexistent_fname = fname .. '.nonexistent'
+    local dollar_fname = '$' .. fname
+    before_each(function()
+      clear()
+      os.remove(fname)
+      os.remove(dollar_fname)
+    end)
+    after_each(function()
+      os.remove(fname)
+      os.remove(dollar_fname)
+    end)
+    it('treats - as stdin', function()
+      eq(nil, lfs.attributes(fname))
+      funcs.system(
+        {nvim_prog_abs(), '-u', 'NONE', '-i', 'NONE', '--headless',
+         '--cmd', 'set noswapfile shortmess+=IFW fileformats=unix',
+         '-s', '-', fname},
+        {':call setline(1, "42")', ':wqall!', ''})
+      eq(0, eval('v:shell_error'))
+      local attrs = lfs.attributes(fname)
+      eq(#('42\n'), attrs.size)
+    end)
+    it('does not expand $VAR', function()
+      eq(nil, lfs.attributes(fname))
+      eq(true, not not dollar_fname:find('%$%w+'))
+      write_file(dollar_fname, ':call setline(1, "100500")\n:wqall!\n')
+      funcs.system(
+        {nvim_prog_abs(), '-u', 'NONE', '-i', 'NONE', '--headless',
+         '--cmd', 'set noswapfile shortmess+=IFW fileformats=unix',
+         '-s', dollar_fname, fname})
+      eq(0, eval('v:shell_error'))
+      local attrs = lfs.attributes(fname)
+      eq(#('100500\n'), attrs.size)
+    end)
+    it('does not crash after reading from stdin in non-headless mode', function()
```

Comment:
Add 

    if helpers.pending_win32(pending) then return end

here, since this test uses termopen().

Reply:
termopen() works on Windows now, so the above comment is outdated. Though reading from stdin on Windows may be broken (https://github.com/neovim/neovim/issues/6890) so maybe it's expedient to  skip it for now.